iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -613,7 +616,7 @@ public boolean canComputeQueryUsingStats(org.apache.hadoop.hive.ql.metadata.Tabl
         }
       }
     }
-    return false;
+    return true;
```

Comment:
In case of delete files, if Iceberg statsSource can not compute query( eg. count(*)) using stats, i think HMS stats can't either. They both come from the same place -- `SnapshotSummary`.

We should not give a wrong impression that HMS can give the accurate stats.

Reply:
even if we do alter table compute stats?

Reply:
Good catch! 
In case of delete files, `analyze table compute stats`  job can get the accurate stats as it will launch tez task to compute the stats.

And after the job `analyze table compute stats`, the HMS stats will be updated & accurate and  `iceberg.hive.keep.stats` will be true, so we can use the HMS stats to optimize the `count `query. 

But if the statsSource is Iceberg & in case of delete files, even we have done the job `analyze table compute stats`, we won't update the Iceberg `SnapshotSummary`, so we can not optimize the `count `query.

This will look a little weird. Users do a job `analyze table compute stats` to update the stats, but they can not optimize the `count `query if the statsSource is Iceberg & in case of delete files.